spread buys evenly throughout window, new default behavior

raedah opened the pull request:
successfully tested this commit on testnet

Spreads user buys evenly throughout window. Will reduce the rush of competing buys that happen when a new window begins. Will reduce cost of ticket prices. Will help to better estimate the next stake difficulty.

raedah commented:
rebased and verified on testnet

raedah commented:
rebased, ready for review

raedah commented:
Updated to take account of expected funds from tickets to be redeemed in the current window. If you have 0.02 proportionlive, you can expect 144 * 5 * 0.02 = ~14.4 ticket funds redeemed in the current window.

alexlyp commented on ticketbuyer/buyer.go:
```diff
@@ -427,9 +467,72 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
+			rand.Seed(time.Now().UTC().UnixNano())
+			log.Debugf("To spend existing funds within this window you need %.3f tickets per block"+
+				", %.3f purchases total", neededPerBlock, toBuyDiffPeriod)
+			log.Tracef("redeemedThisWindow vars", winSize, t.idxDiffPeriod, t.activeNet.TicketsPerBlock, t.ProportionLive)
```
Need to add %v (or %.3f etc)

raedah replied:
just removed it instead

raedah commented:
Reformatted code for readability, cleaned up variable names, added comments, and did another read through. Running this locally on testnet and is working well.

raedah commented on ticketbuyer/purchase.go:
```diff
@@ -31,7 +31,6 @@ func NewPurchaseManager(w *wallet.Wallet, purchaser *TicketPurchaser,
 
 // purchase purchases the tickets for the given block height.
 func (p *PurchaseManager) purchase(height int64) {
-	log.Infof("Block height %v connected", height)
```
This was removed because it is redundant with dcrwallets output. Was only useful when ticketbuyer was its own app.

raedah commented on ticketbuyer/buyer.go:
```diff
@@ -199,61 +211,65 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
-	// different window period. If this is the case,
-	// we need reset our variables too.
-	thisWindowPeriod := int(height / winSize)
-	if (height+1)%winSize != 0 &&
```
this +1 here was erroneous, and is not used above at my line 240

raedah commented on ticketbuyer/buyer.go:
```diff
@@ -199,61 +211,65 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
-			log.Errorf("Failed to decode tx fee amount %v from config",
-				t.cfg.TxFee)
-		} else {
-			t.wallet.SetRelayFee(txFeeAmt)
```
setting the txfee is no longer needed, as this was old workaround behavior for old defaults

raedah commented on ticketbuyer/buyer.go:
```diff
@@ -310,36 +329,34 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
 
 	// Scale the average price according to the configuration parameters
 	// to find minimum and maximum prices for users that are electing to
 	// attempting to manipulate the stake difficulty.
```
below, just moved the code so that min comes before max
